browser/components/CodeEditor.js

```diff
@@ -746,6 +775,12 @@ export default class CodeEditor extends React.Component {
     }
   }
 
+  linePossibleContainsHeadline (currentLine) {
+    // We can't check if the line start with # because when some write text before
+    // the # we also need to update the toc
+    return currentLine.includes('# ')
```

Comment:
Why can't we? We can just replace it with `'\n# '` don't you think?

Reply:
I didn't that matching '\n# ' helps. The line starts with the first char that is't a new line.
And we should check for the '# ' because when you add to a headline with '## test'  something at the beginning (new text is: 'something## test') the toc should automatic remote the entry

Reply:
@ZeroX-DG

Reply:
You're right!